src/reader.cpp

```diff
@@ -3109,11 +3112,14 @@ void reader_data_t::handle_readline_command(readline_cmd_t c, readline_loop_stat
             editable_line_t *el = active_edit_line();
             if (is_navigating_pager_contents()) {
                 select_completion_in_direction(selection_motion_t::east);
+                inputter.function_set_status(true);
             } else if (el->position() < el->size()) {
                 update_buff_pos(el, el->position() + 1);
                 mark_repaint_needed();
+                inputter.function_set_status(true);
             } else {
-                accept_autosuggestion(true);
+                accept_autosuggestion(false, true);
+                inputter.function_set_status(false);
```

Comment:
Ugh.

Yeah, that makes sense. Guess it's back to the drawing board then.

Reply:
We could expose a readline command that tells fish to coalesce the next edit with the previous undo group.
That would be useful if we have another similar binding. Let's check that, otherwise it's probably not worth it.

Reply:
Outside vi bindings that seems to be only the keys bound to `self-insert expand-abbr`, I guess for those it's alright that `undo` reverts them separately.

Reply:
Either way, too late. I already added a forward_single_char command.